src/main/java/com/hcl/appscan/jenkins/plugin/scanners/DynamicAnalyzer.java

```diff
@@ -129,46 +138,117 @@ public void setOptimization(String optimization) {
 			m_optimization = optimization;
 		}
 	}
-	
+
 	public String getOptimization() {
-	    m_optimization = mapOldtoNewOptLevels(m_optimization);
+		m_optimization = mapOldtoNewOptLevels(m_optimization);
 		return m_optimization;
 	}
-	
+
 	@DataBoundSetter
 	public void setExtraField(String extraField) {
 		m_extraField = extraField;
 	}
-	
+
 	public String getExtraField() {
 		return m_extraField;
 	}
-	
+
+	@DataBoundSetter
+	public void setLoginType(String loginType) {
+			m_loginType =loginType;
+	}
+
+	public String getLoginType() {
+		return m_loginType;
+	}
+
+	@DataBoundSetter
+	public void setTrafficFile(String trafficFile) {
+		if (RECORDED.equals(m_loginType))
+			m_trafficFile = trafficFile;
+	}
+
+	public String getTrafficFile() {
+		return m_trafficFile;
+	}
+
 	@Override
 	public String getType() {
 		return DYNAMIC_ANALYZER;
 	}
-	
+
+	public String isLoginTypes(String loginTypeName) {
+		if (m_loginType != null) {
+			return m_loginType.equalsIgnoreCase(loginTypeName) ? "true" : "";
+		} else if((m_loginUser != null || m_loginPassword !=null)){
+			m_loginType = AUTOMATIC;
+			return "true";
+		} else if (loginTypeName.equals(NONE)) { //Default
+			return "true";
+		} else {
+			return "";
+		}
+	}
+
+	public String upgradeLoginScenario(){
+		if(m_loginUser != null || m_loginPassword != null){
+			return m_loginType = AUTOMATIC;
+		} else {
+			return m_loginType = NONE;
+		}
+	}
+
 	@Override
-	public Map<String, String> getProperties(VariableResolver<String> resolver) {
+	public Map<String, String> getProperties(VariableResolver<String> resolver) throws hudson.AbortException {
 		Map<String, String> properties = new HashMap<String, String>();
-                if(resolver == null) {
+		if (resolver == null) {
 			properties.put(TARGET, getTarget());
-			properties.put(LOGIN_USER, m_loginUser);
-			properties.put(LOGIN_PASSWORD, Secret.toString(m_loginPassword));
 			properties.put(SCAN_FILE, m_scanFile);
 			properties.put(EXTRA_FIELD, m_extraField);
-                }
-                else {
+			if(m_loginType == null){
+				m_loginType = upgradeLoginScenario();
+			}
+				if (Objects.equals(m_loginType, RECORDED)) {
```

Comment:
Why is `Objects.equals()` being used here instead of just comparing the string values with `RECORDED.equals(m_loginType)`?